### PR TITLE
Add visuals and diagrams to ug

### LINKED ISSUE #116: [PE-D] No list of valid file extensions for diagram
As a user, I have to do trial and error to find what kind of files can be saved as diagrams. Neither the error message nor the user guide specifies the files extensions that can be saved.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: ZoroarkDarkrai/ped#4

### LINKED ISSUE #121: [PE-D] UserGuide no use of visuals
Can consider add some screenshots, your UI is very impressive.

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: yuanxing-y/ped#2

### LINKED ISSUE #136: [PE-D] No pictures shown in the UG
UG can include some pictures to assist users

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: juandavinlie/ped#3

### LINKED ISSUE #137: [PE-D] Diagrams hard to test as user
It may be useful to include guides on how to upload a diagram and retrieve its file path.

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: juandavinlie/ped#4

### LINKED ISSUE #140: [PE-D] Hyperlinks can be added to imporve navigation
-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: juandavinlie/ped#7

### LINKED ISSUE #144: [PE-D] Expected outcome/behaviour is not shown in the UG
Expected outcome may be beneficial to users as it reassures them that their inputs are correct

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: juandavinlie/ped#11

### LINKED ISSUE #161: Add disclaimer for no spaces in diagram files paths in UG

### PR DESCRIPTION
**Pull request by moyj01**

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -7,26 +7,27 @@ SWEe! is a  **desktop app for CS2103T students to manage their learning progress
...
-        - Exiting the program: `exit`
-        - View a flashcard: `view`
-        - Saving the data
+    - [Quick start](#quick-start)
+    - [Features](#features)
+        - [Adding a flashcard](#adding-a-flashcard--add): `add`
+        - [Clear all flashcards](#clearing-all-flashcards--clear): `clear`
+        - [Deleting a flashcard](#deleting-a-flashcard---delete): `delete`
+        - [Editing a flashcard](#editing-a-flashcard---edit): `edit`
+        - [Filter relevant flashcards](#filtering-out-flashcards---filter): `filter`
```
Comment (sgn00) [inline docs/UserGuide.md]: Standardise naming. Either filter relevant or filtering out flashcards.

File: docs/UserGuide.md [hunk 2]
```diff
@@ -7,26 +7,27 @@ SWEe! is a  **desktop app for CS2103T students to manage their learning progress
...
+        - [Favourite a flashcard](#favourite-a-flashcard---fav) : `fav`
+        - [Unfavourite a flashcard](#unfavourite-a-flashcard---unfav): `unfav`
+        - [Find flashcards](#find-flashcards--find): `find`
+        - [Viewing help](#viewing-help--help): `help`
+        - [Listing all flashcards](#listing-all-flashcards--list): `list`
+        - [Review all flashcards](#reviewing-all-flashcards-review) : `review`
```
Comment (sgn00) [inline docs/UserGuide.md]: Note we are not reviewing *all* flashcards. Change to review flashcards.

File: docs/UserGuide.md [hunk 3]
```diff
@@ -7,26 +7,27 @@ SWEe! is a  **desktop app for CS2103T students to manage their learning progress
...
+        - [Quiz flashcards](#quizzing-flashcards-quiz): `quiz`
+        - [Sort all flashcards](#sort-all-flashcards--sort): `sort`
```
Comment (sgn00) [inline docs/UserGuide.md:24]: Remove `all`

File: docs/UserGuide.md [hunk 4]
```diff
@@ -167,6 +192,20 @@ Examples:
...
+
+**Step 1**: Locate the flashcard you wish to edit. In this example, we want to edit the flashcard at index 2.
+
+![filedirectory](images/ug/ug_edit_step1.PNG)
+
+**Step 2**: We want to edit the category of the flashcard to Trivial and rating to 1. Type the command `edit 2 c/Trivial r/1`.
+
+![filedirectory](images/ug/ug_edit_step2.PNG)
+
+**Step 3**: The flashcard's category and rating are edited to `Trivial` and `1` respectively.
```
Comment (sgn00) [inline docs/UserGuide.md]: Remove code highlight.

File: docs/UserGuide.md [hunk 5]
```diff
@@ -167,6 +192,20 @@ Examples:
 * `edit 3 q/What does OOP stand for? a/Object Oriented Programming`
 * `edit 3 n/Important question! r/`
 
+**Steps for editing a flashcard's category and rating**:
+
+**Step 1**: Locate the flashcard you wish to edit. In this example, we want to edit the flashcard at index 2.
+
+![filedirectory](images/ug/ug_edit_step1.PNG)
+
+**Step 2**: We want to edit the category of the flashcard to Trivial and rating to 1. Type the command `edit 2 c/Trivial r/1`.
```
Comment (sgn00) [inline docs/UserGuide.md]: Remove fullstop at the end

File: docs/UserGuide.md [hunk 6]
```diff
@@ -185,6 +224,15 @@ Examples:
 *  `filter t/examinable t/study` filters and lists all flashcards that have both “examinable” and “study”.
 *  `filter r/3 f/yes` filters and lists all favourited flashcards that have a rating of 3.
 
+**Steps for filtering for a flashcard based on category and tag**:
+
+**Step 1**: We want to filter for a flashcard which has a Trivial category and contains the preloaded tag. Type the command `filter c/Trivial t/preloaded`.
+
+![filedirectory](images/ug/ug_filter_step1.PNG)
+
+**Step 2**: The flashcard with the category `Trivial` and tag field `preloaded` is shown.
```
Comment (sgn00) [inline docs/UserGuide.md]: Remove code highlight.

File: docs/UserGuide.md [hunk 7]
```diff
@@ -270,13 +333,35 @@ Upon entering quiz mode, the following user input will be recognised:
...
 </div>
+<br>
+
+**Steps for entering quiz mode**:
+
+**Step 1**: We want to enter quiz mode to test ourselves on the current list of flashcards. Type the command `quiz`.
+
+![filedirectory](images/ug/ug_quiz_step1.PNG)
+
+**Step 2**: We are brought to the quiz mode. The instructions on how to navigate quiz mode will be shown at the top.
```
Comment (sgn00) [inline docs/UserGuide.md]: We are brought into quiz mode.

File: docs/UserGuide.md [hunk 8]
```diff
@@ -301,12 +386,67 @@ Format: `view INDEX [-a]`
...
+
+**Step 1**: Locate the flashcard you wish to view. In this example, we want to view the flashcard at index 3. Type the command `view 3`.
+
+![filedirectory](images/ug/ug_view_step1.PNG)
+
+**Step 2**: We will be presented with a "snapshot" of the flashcard at index 3 in the view pane.
+
+![filedirectory](images/ug/ug_view_step2.PNG)
+
+**Step 3**: To view the answer and notes(if applicable) of the flashcard on the view pane. Type the command `view 3 -a`.
```
Comment (sgn00) [inline docs/UserGuide.md]: Missing space notes(if applicable)

File: docs/UserGuide.md [hunk 9]
```diff
@@ -7,26 +7,27 @@ SWEe! is a  **desktop app for CS2103T students to manage their learning progress
...
-        - Sort all flashcards: `sort`
-        - Exiting the program: `exit`
-        - View a flashcard: `view`
-        - Saving the data
+    - [Quick start](#quick-start)
+    - [Features](#features)
+        - [Adding a flashcard](#adding-a-flashcard--add): `add`
+        - [Clear all flashcards](#clearing-all-flashcards--clear): `clear`
+        - [Deleting a flashcard](#deleting-a-flashcard---delete): `delete`
+        - [Editing a flashcard](#editing-a-flashcard---edit): `edit`
```
Comment (sgn00) [inline docs/UserGuide.md]: Standardise present continuous or present tense.

**Review by sgn00 — CHANGES_REQUESTED** [review]
Refer to comments.

File: docs/UserGuide.md (hunk 1 above)
Comment (moyj01) [reply docs/UserGuide.md]: Will standardise to present tense. Thank you for pointing it out.

File: docs/UserGuide.md (hunk 2 above)
Comment (moyj01) [reply docs/UserGuide.md]: Thanks for pointing it out. Will do 👍

File: docs/UserGuide.md (hunk 3 above)
Comment (moyj01) [reply docs/UserGuide.md:24]: Removed accordingly.

File: docs/UserGuide.md (hunk 4 above)
Comment (moyj01) [reply docs/UserGuide.md]: My bad. Removed.

File: docs/UserGuide.md (hunk 5 above)
Comment (moyj01) [reply docs/UserGuide.md]: Removed

File: docs/UserGuide.md (hunk 6 above)
Comment (moyj01) [reply docs/UserGuide.md]: Same as the issue before. Rectified

File: docs/UserGuide.md (hunk 7 above)
Comment (moyj01) [reply docs/UserGuide.md]: Corrected the language

File: docs/UserGuide.md (hunk 8 above)
Comment (moyj01) [reply docs/UserGuide.md]: Added a space

File: docs/UserGuide.md (hunk 9 above)
Comment (moyj01) [reply docs/UserGuide.md]: Standardised all to present tense less saving the data.

File: docs/UserGuide.md [hunk 10]
```diff
@@ -257,8 +305,23 @@ Format: `review`
...
+**Step 1**: We want to enter review mode to review our flashcards. Type the command `review`
+
+![filedirectory](images/ug/ug_review_step1.PNG)
+
+**Step 2**: We are brought into review mode. The instructions on how to navigate review mode will be shown at the top.
+
+![filedirectory](images/ug/ug_review_step2.PNG)
 
-### Quizzing flashcards: `quiz`
+**Step 3**: In this example, we will demonstrate the behaviour of the `↓ key`. Upon clicking the `↓ key`, the answer of the flashcard is shown.
```
Comment (sgn00) [inline docs/UserGuide.md]: "pressing" not "clicking"

File: docs/UserGuide.md [hunk 11]
```diff
@@ -270,17 +333,39 @@ Upon entering quiz mode, the following user input will be recognised:
...
+
+**Step 1**: We want to enter quiz mode to test ourselves on the current list of flashcards. Type the command `quiz`
+
+![filedirectory](images/ug/ug_quiz_step1.PNG)
+
+**Step 2**: We are brought into quiz mode. The instructions on how to navigate quiz mode will be shown at the top.
+
+![filedirectory](images/ug/ug_quiz_step2.PNG)
+
+**Step 3**: In this example, we will demonstrate the behaviour of the `↓ key`. Upon clicking the `↓ key`, we will be prompted if we got the answer correct.
```
Comment (sgn00) [inline docs/UserGuide.md]: "pressing" not "clicking"

**Review by sgn00 — CHANGES_REQUESTED** [review]
Minor change.

File: docs/UserGuide.md (hunk 10 above)
Comment (moyj01) [reply docs/UserGuide.md]: Ammended

File: docs/UserGuide.md (hunk 11 above)
Comment (moyj01) [reply docs/UserGuide.md]: Amended

**Review by sgn00 — APPROVED** [review]

**Pull request merged**